homework/index.js

```diff
@@ -29,23 +31,160 @@
     return elem;
   }
 
+  function convertDateTime(dateTimeTxt) {
+    const dateTime = new Date(dateTimeTxt);
+    return dateTime.toLocaleString();
+  }
+
+  function sortRepositoriesByName(a, b) {
+    return a.name.localeCompare(b.name);
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    ul.textContent = ' ';
+    const repoList = createAndAppend('li', ul, { class: 'rep-list' });
+    const repositoryName = createAndAppend('p', repoList, {
+      text: ' Repository: ',
+      class: 'repo-details',
+    });
+
+    const description = createAndAppend('p', repoList, {
+      text: ' Description: ',
+      class: 'repo-details',
+    });
+
+    const forkedNumbers = createAndAppend('p', repoList, {
+      text: ' Forks: ',
+      class: 'repo-details',
+    });
+
+    const updatedTime = createAndAppend('p', repoList, {
+      text: ' Updated: ',
+      class: 'repo-details',
+    });
+
+    createAndAppend('a', repositoryName, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+
+    if (repo.description === null) {
+      createAndAppend('span', description, {
+        text: 'No Description Added Yet',
+        class: 'texts',
+      });
+    } else {
+      createAndAppend('span', description, {
+        text: repo.description,
+        class: 'texts',
+      });
+    }
```

Comment:
Rather than repeating almost identical code in the `if` and `else` parts, you can use a so-called _short-circuit_ expression to deal with a missing description:

```js
createAndAppend('span', description, {
  text: repo.description || 'No Description Added Yet',
  class: 'texts',
});
```

The `text` attribute will be set to `repo.description` if it is truthy (a non-empty string is truthy), otherwise it will be set to `No Description Added Yet`.

Reply:
I changed the rewrote the codes based on your feedback, and it is better now. I both save space and the codes looks nicer now 👍